CS updated as per camel 3.0.0 specs

varunblaze opened the pull request.

varunblaze commented:
@oscerd , Please review the pull request.  Thanks.

oscerd commented:
Once I have the time

dmvolod commented:
@varunblaze I think it's better to cherry-pick component commits from camel-2.x branch and fix Camel 3.0 difference rather thank create component from scratch for 3.0 version.

oscerd commented:
It's not on 2.x. the first released version will be with 3.0.0 that's why I told him to open a PR against master

dmvolod commented:
@oscerd oops, sorry, missing this, but PR name is a little bit confusing...

oscerd commented:
No problem :-) this PR has been open for a while, while we were switching to 3.x, after 2.23.0 release, the PR was messed up, so we decided to move to 3.x directly

varunblaze commented:
Hi,

I have incorporated all the camel. X changes as suggested and updated the
version accordingly.

Thanks
Varun

On Sun, 6 Jan 2019, 2:34 pm Andrea Cosentino <notifications@github.com
wrote:

> No problem :-) this PR has been open for a while, while we were switching
> to 3.x, after 2.23.0 release, the PR was messed up, so we decided to move
> to 3.x directly
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/camel/pull/2688#issuecomment-451726343>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADN7BT5jt9gE1rtIowsj-mNzALr61aPqks5vAbwEgaJpZM4Zg015>
> .
>

dmvolod commented on components/camel-chatscript/src/test/resources/log4j.properties:
```diff
@@ -0,0 +1,33 @@
+## ---------------------------------------------------------------------------
```
Please remove file, we are not using this logger

varunblaze replied:
done

dmvolod commented on components/camel-chatscript/src/main/docs/camel-chatscript-component.adoc:
```diff
@@ -0,0 +1,77 @@
+[[ChatScript-component]]
```
adoc file starting with camel- is not correct, please remove one of doc files

varunblaze replied:
done

dmvolod commented on components/camel-chatscript/src/main/java/org/apache/camel/component/chatscript/ChatScriptEndpoint.java:
```diff
@@ -0,0 +1,138 @@
+    private int port;
+    @UriPath(description = "Name of the Bot in CS to converse with")
+    @Metadata(required = "true")
+    private String botname;
```
The "CamelCase" can be used for UriPath parameters also

varunblaze replied:
done

dmvolod commented on components/camel-chatscript/src/main/java/org/apache/camel/component/chatscript/ChatScriptEndpoint.java:
```diff
@@ -0,0 +1,138 @@
+            ChatScriptComponent component) throws URISyntaxException {
+        super(uri, component);
+
+        URI remainingUri = new URI("tcp://" + remaining);
```
Is starting with 'tcp' is correct?

varunblaze replied:
ChatScript URI is exactly same as the URI for tcp but the fact that the component name will be different.

dmvolod commented on components/camel-chatscript/src/main/java/org/apache/camel/component/chatscript/ChatScriptEndpoint.java:
```diff
@@ -0,0 +1,138 @@
+ */
+@UriEndpoint(firstVersion = "3.0.0", scheme = "chatscript", title = "ChatScript", syntax = "chatscript:host:port/botname",  producerOnly = true, label = "ai,chatscript")
+public class ChatScriptEndpoint extends DefaultEndpoint { 
+    private static final String URI_ERROR = "Invalid URI. Format must be of the form chatscript://host[:port]/botname?[options...]";
```
Could be moved to *Constants instead of placed inside *Endpoint

varunblaze replied:
done

oscerd commented:
I'll wait for the fixes first.

oscerd commented:
Thanks. I'll merge ASAP

oscerd commented:
I'd like first to clean up the codebase of the master branch and then I'll work on this.

varunblaze commented:
@oscerd , Please let me know if I need to work on resolving the conflict.

oscerd commented:
Yes. If you can it would be great. Sorry for the delay but we have a bit more stable codebase now

varunblaze commented:
@oscerd, Sure, will do it later today and keep you posted.

varunblaze commented:
Hi @oscerd, have updated the pom files. Can you please mark the conflict as resolved and merge?

varunblaze commented:
@oscerd  Any update?

oscerd commented:
I'm merging. There are some errors and I'm adding the stuff to the kit.

oscerd commented:
Thanks, this has been merged.

varunblaze commented:
Thanks @oscerd...
